src/utils/api_request_helpers.ts

```diff
@@ -0,0 +1,124 @@
+import { error } from 'console'
+import dotenv from 'dotenv'
+
+// Load environment variables from .env file
+dotenv.config()
+
+//Interfaces
+
+export interface Racemap_event {
+    name: string
+    r_id: string
+}
+
+export interface Event_shadow_coordinates {
+    event_id: string
+    lat: number
+    lng: number
+}
+
+// Function that creates an auth header for requests
+export const generate_authentication_header = async (): Promise<string> => {
+    // Retrieve the environment variable
+    const user_email = process.env.USER_EMAIL
+    const user_password = process.env.USER_PASSWORD
+
+    // Merging the environment variables into an Authorization header to send with the request
+    // Exporting that it can be used in various requests
+    // "Authorization": `Basic ${credentials}`
+    const credentials = Buffer.from(`${user_email}:${user_password}`, 'utf-8')
+
+    //Convert the Buffer to a Base64-encoded string
+    const base64Credentials = credentials.toString('base64')
+
+    console.log('Creditals: ' + base64Credentials)
+    return base64Credentials
+}
+
+// Function that returns the data given from any API endpoint --> Multiple usecases
+const return_data_from_api_endpoint = async (
+    url: string
+): Promise<Response> => {
+    let token: string = await generate_authentication_header()
+    let header: Headers = new Headers({
+        Authorization: `Basic ${token}`,
+    })
+    const response_data = await fetch(url, { method: 'GET', headers: header })
+    // response_data includes the status, headers and the body
+    return response_data
+}
+
+// Function that uses return_data_from_api_endpoint to collect
+export const return_entirety_of_events = async (): Promise<Racemap_event[]> => {
+    // Setting up a dictionary that maps the names of the events with corresponding ids
+    let entirety_of_events: Racemap_event[] = []
+
+    try {
+        // Calling the async function that returns a Promise<Response>
+        const response = await return_data_from_api_endpoint(
+            process.env.ALL_EVENTS_URL_ENDPOINT as string
+        )
+
+        if (!response.ok) {
+            console.log('Network response was not okay (return event_id, name)')
+            return []
+        }
+
+        const data = await response.json()
+
+        // Looping through the response body
+        // Filtering the large response body
+        for (const index of data) {
+            entirety_of_events.push({ name: index.name, r_id: index.id })
```

Comment:
Alternativly you can store the information in a Map. Then you can access the name later faster per event id.